### DISCUSSION TITLE
the content of the file  which is uploaded to the "kowledge" can be read by the webui

### DISCUSSION THREAD

**Discussion by Milton113041** [open-webui/open-webui#6480] [Issues]
I have uploaded some files which file type is "doc" and "docx" to the "kowledge", but the content cann‘t be read by the webui, It appears below:
 "{
    "detail": "[ERROR: Error getting file content]"
}"

**Comment by whichee** [comment]
The error you're encountering is due to the filename containing characters that are not supported by the "latin-1" codec. The command-line output shows:

```python
(k.lower().encode("latin-1"), v.encode("latin-1"))
                                  ^^^^^^^^^^^^^^^^^^^
UnicodeEncodeError: 'latin-1' codec can't encode characters in position 22-25: ordinal not in range(256)
ERROR [open_webui.apps.webui.routers.files] Error getting file content
INFO:     127.0.0.1:6478 - "GET /api/v1/files/7c231ba8-4d50-4cb1-9397-96e3453e1042/content HTTP/1.1" 400 Bad Request
```

To resolve this issue, you can either rename the uploaded file to use only ASCII characters or modify the code to handle UTF-8 encoding. For example, in the `\Lib\site-packages\starlette\responses.py` file, you can change the following line:

```python
raw_headers = [
    (k.lower().encode("utf-8"), v.encode("utf-8"))
]
```

This modification will ensure that the headers are encoded using UTF-8 instead of "latin-1".

**Comment by chen2aaron** [comment]
https://github.com/open-webui/open-webui/pull/7807 will fix it.
